### PR TITLE
Admin client

### PR DESCRIPTION
**Pull request by spolu**
Simple admin client:
```
spolu@spolu-dev ~/code/dust/front (spolu-admin_cli) $ ./admin/cli.sh user show --userId 1
user:
  id: 1
  username: spolu
  name: Stanislas Polu
  email: spolu@proton.me
  workspaces:
    - wId: bc9d30067b
      name: spolu
      role: admin
```

### CODE REVIEW

File: front/admin/cli.ts [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+        where: {
+          sId: args.wId,
+        },
+      });
+
+      if (!w) {
+        throw new Error(`Workspace not found: wId='${args.wId}'`);
+      }
+
+      console.log(`workspace:`);
```
Comment (lasryaric) [inline front/admin/cli.ts:43]: You could define a little interface and pretty print it if it makes your life easier:

```
interface adultWithKids {
  name:  string;
  age: number;
  kids: {
    name: string;
    age: number;
    sex: string;
  }[]
}

const arico: adultWithKids = {
  name:"aric lasry",
  age: 32,
  kids: [
    {
      name: "milo lasry",
      age: 2,
      sex:"M" 
    },
    {
      name: "luna lasry",
      age: 2,
      sex:"F"
    }
  ]
}

console.log(JSON.stringify(arico, null, 2))
```

File: front/admin/cli.ts [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+    }
+    case "change-role": {
+      if (!args.wId) {
+        throw new Error("Missing --wId argument");
+      }
+      if (!args.userId) {
+        throw new Error("Missing --userId argument");
+      }
+      if (!args.role) {
+        throw new Error("Missing --userId argument");
```
Comment (lasryaric) [inline front/admin/cli.ts:198]:
```suggestion
        throw new Error("Missing --role argument");
```

**Review by lasryaric — CHANGES_REQUESTED** [review]
[commander.js](https://github.com/tj/commander.js/) looks like a good CLI helper to do all the parsing and matching if you wanted to make it easier to read / maintain.
Otherwise I posted a few small comments to address.

In any case, that was fast! 🔥

File: front/admin/cli.ts (hunk 1 above)
Comment (lasryaric) [reply front/admin/cli.ts:43]:
```
{
  "name": "aric lasry",
  "age": 32,
  "kids": [
    {
      "name": "milo lasry",
      "age": 2,
      "sex": "M"
    },
    {
      "name": "luna lasry",
      "age": 2,
      "sex": "F"
    }
  ]
}
```

**Comment by spolu** [comment]
Merging as is, let's explore more elaborate parsing strategies that support subcommand as this file grows :+1:

**Pull request merged**